survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/question_parts/LocationPickerPart.kt

```diff
@@ -0,0 +1,268 @@
+        search.setHint(android.R.string.search_go)
+        search.setSingleLine(true)
+        search.setCompoundDrawables(
+            resources.getDrawable(R.drawable.ic_baseline_search_24),
+            null,
+            null,
+            null
+        )
+        search.compoundDrawablePadding = 32
+
+        mapView.getMapAsync { map ->
+            this.map = map
+            if (::mapReadySelection.isInitialized) {
+                selected = mapReadySelection
+            }
+
+            if (ActivityCompat.checkSelfPermission(
+                    mapView.context,
+                    Manifest.permission.ACCESS_FINE_LOCATION
+                ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+                        mapView.context,
+                        Manifest.permission.ACCESS_COARSE_LOCATION
+                    ) != PackageManager.PERMISSION_GRANTED
+            ) {
+                // TODO ask location permission
+                return@getMapAsync
+            }
+            this.map.isMyLocationEnabled = true
+            this.map.uiSettings.isMyLocationButtonEnabled = true
+
+            this.map.setOnMyLocationChangeListener { location ->
+                if (selected.latitude == 0.0 && selected.longitude == 0.0) {
+                    this.map.moveCamera(
+                        CameraUpdateFactory.newLatLngZoom(
+                            LatLng(
+                                location.latitude,
+                                location.longitude
+                            ),
+                            15f
+                        )
+                    )
+                }
+            }
+
+            this.map.setOnCameraMoveStartedListener { reason ->
+                if (reason == GoogleMap.OnCameraMoveStartedListener.REASON_GESTURE) {
+                    locationMarker.animate().translationY(-10f)
+                        .setInterpolator(BounceInterpolator()).setDuration(300)
+                        .start()
+                }
+            }
+
+            this.map.setOnCameraIdleListener {
+                locationMarker.animate().translationY(10f)
+                    .setInterpolator(BounceInterpolator()).setDuration(300)
+                    .start()
+            }
+        }
+        lifecycle.addObserver(this)
+
+        this.addView(search)
+
+        locationMarker.setImageResource(R.drawable.ic_baseline_location_on)
+        frame.addView(mapView, LayoutParams(LayoutParams.MATCH_PARENT, 800))
+        val locationMarkerParams =
+            FrameLayout.LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT)
+        locationMarkerParams.gravity = Gravity.CENTER
+        frame.addView(locationMarker, locationMarkerParams)
+
+        this.addView(frame, LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT))
+
+        search.threshold = 4
+        search.addTextChangedListener(object : TextWatcher {
```

Comment:
There exists a custom extension function which does exactly this `afterTextChanged`. 
```kotlin
        search.afterTextChanged { s ->
            s.let {
                if (!search.isPerformingCompletion) {
                    addressProvider.input(s)
                }
            }
        }
```